csharp/src/Apache.Arrow/Extensions/SpanExtensions.cs

```diff
@@ -14,18 +14,65 @@
 // limitations under the License.
 
 using System;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
 namespace Apache.Arrow
 {
     public static class SpanExtensions
     {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> CastTo<T>(this Span<byte> span)
             where T: struct =>
             MemoryMarshal.Cast<byte, T>(span);
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ReadOnlySpan<T> CastTo<T>(this ReadOnlySpan<byte> span)
             where T: struct =>
                 MemoryMarshal.Cast<byte, T>(span);
+
+
```

Comment:
Can we add a comment here saying `TODO remove these methods once dotnet/coreclr#27590 is fixed`?